broker/service/api/v10.py

```diff
@@ -53,16 +53,17 @@ def run_submission(data):
             API_LOG.log("Unauthorized request")
             raise ex.UnauthorizedException()
 
-    else:
-        if data['plugin'] not in api.plugins:
-            raise ex.BadRequestException()
-        plugin = plugin_base.PLUGINS.get_plugin(data['plugin'])
-        submission_data = data['plugin_info']
-        submission_data['enable_auth'] = data['enable_auth']
-        submission_id, executor = plugin.execute(submission_data)
-        submissions[submission_id] = executor
+    if data['plugin'] not in api.plugins:
+        API_LOG.log("Plugin is missing")
```

Comment:
improve this message. tell the user which plugin failed (`data['plugin']`) and all the plugins that are currently available (from `api.plugins`).

Reply:
same for the exception below. (why logging it before if we're raising the exception anyway?).

Reply:
I actually just changed the message to be more informative, the reason it is logginhg before raising the exception it's because all other exceptions are doing this way, i changed it so the code could be more cohesive